File: Strings/1873/1873.py
def rajesh(r,s):
    if r == "tesoura" and (s == "papel" or s == "lagarto"):
        return True
    elif r == "papel" and (s == "pedra" or s == "spock"):
        return True
    elif r == "pedra" and (s == "lagarto" or s == "tesoura"):
        return True
    elif r == "lagarto" and (s == "spock" or s == "papel"):
        return True
    elif r == "spock" and (s == "tesoura" or s == "pedra"):
        return True
    return False

def sheldon(r,s):
    if s == "tesoura" and (r == "papel" or r == "lagarto"):
        return True
    elif s == "papel" and (r == "pedra" or r == "spock"):
        return True
    elif s == "pedra" and (r == "lagarto" or r == "tesoura"):
        return True
    elif s == "lagarto" and (r == "spock" or r == "papel"):
        return True
    elif s == "spock" and (r == "tesoura" or r == "pedra"):
        return True
    return False

n = int(input())
for i in range(n):
    r,s = input().split()
    if rajesh(r,s):
        print("rajesh")
    elif sheldon(r,s):
        print("sheldon")
    else:
        print("empate")

File: Strings/1873/1873.c
#include <stdio.h>
#include <string.h>

int rajesh(char *r, char *s) {
    if (strcmp(r, "tesoura") == 0 && (strcmp(s, "papel") == 0 || strcmp(s, "lagarto") == 0)) return 1;
    if (strcmp(r, "papel") == 0 && (strcmp(s, "pedra") == 0 || strcmp(s, "spock") == 0)) return 1;
    if (strcmp(r, "pedra") == 0 && (strcmp(s, "lagarto") == 0 || strcmp(s, "tesoura") == 0)) return 1;
    if (strcmp(r, "lagarto") == 0 && (strcmp(s, "spock") == 0 || strcmp(s, "papel") == 0)) return 1;
    if (strcmp(r, "spock") == 0 && (strcmp(s, "tesoura") == 0 || strcmp(s, "pedra") == 0)) return 1;
    return 0;
}

int sheldon(char *r, char *s) {
    if (strcmp(s, "tesoura") == 0 && (strcmp(r, "papel") == 0 || strcmp(r, "lagarto") == 0)) return 1;
    if (strcmp(s, "papel") == 0 && (strcmp(r, "pedra") == 0 || strcmp(r, "spock") == 0)) return 1;
    if (strcmp(s, "pedra") == 0 && (strcmp(r, "lagarto") == 0 || strcmp(r, "tesoura") == 0)) return 1;
    if (strcmp(s, "lagarto") == 0 && (strcmp(r, "spock") == 0 || strcmp(r, "papel") == 0)) return 1;
    if (strcmp(s, "spock") == 0 && (strcmp(r, "tesoura") == 0 || strcmp(r, "pedra") == 0)) return 1;
    return 0;
}

int main(void) {
    int n;
    scanf("%d", &n);
    char r[8], s[8];

    for (int i = 0; i < n; i++) {
        scanf("%s %s", r, s); 
        if (rajesh(r, s)) printf("rajesh\n");
        else if (sheldon(r, s)) printf("sheldon\n");
        else printf("empate\n");
    }
    return 0;
}
